From 6053c67da81b94d6e74a8f55cba9e6873bf4e571 Mon Sep 17 00:00:00 2001
From: Stefan Vigerske <svigerske@gams.com>
Date: Fri, 22 Jul 2022 11:36:15 +0200
Subject: [PATCH] make sure _Thread_local is also defined for C >= 11

- remove unnecessary handling for older GCC
  (for any GCC, _Thread_local is defined to __thread anyway)
---
 src/tinycthread/tinycthread.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/source/tinycthread.h b/source/tinycthread.h
--- a/source/tinycthread.h
+++ b/source/tinycthread.h
@@ -150,14 +150,14 @@ int _tthread_timespec_get(struct timespec *ts, int base);
 * @hideinitializer
 */
 
-#if !(defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201102L)) && !defined(_Thread_local)
- #if defined(__GNUC__) || defined(__INTEL_COMPILER) || defined(__SUNPRO_CC) || defined(__IBMCPP__)
+#if !defined(_Thread_local)
+ #if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201102L
+  #define _Thread_local _Thread_local
+ #elif defined(__GNUC__) || (defined(__INTEL_COMPILER) && !defined(_MSC_VER)) || defined(__SUNPRO_CC) || defined(__IBMCPP__)
   #define _Thread_local __thread
  #else
   #define _Thread_local __declspec(thread)
  #endif
-#elif defined(__GNUC__) && defined(__GNUC_MINOR__) && (((__GNUC__ << 8) | __GNUC_MINOR__) < ((4 << 8) | 9))
- #define _Thread_local __thread
 #endif
 
 /* Macros */
